lib/libp2p_port.ex

```diff
@@ -549,46 +550,42 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
   end
 
   @impl GenServer
-  def handle_call(:get_keystores, _from, %{validators: validators} = state),
-    do: {:reply, Enum.map(validators, fn {_pubkey, validator} -> validator.keystore end), state}
+  def handle_call(:get_keystores, _from, %{validator_set: validator_set} = state),
+    do:
+      {:reply,
+       Enum.map(validator_set.validators, fn {_index, validator} -> validator.keystore end),
+       state}
 
   @impl GenServer
-  def handle_call({:delete_validator, pubkey}, _from, %{validators: validators} = state) do
-    case Map.fetch(validators, pubkey) do
-      {:ok, validator} ->
-        Logger.warning("[Libp2pPort] Deleting validator with index #{inspect(validator.index)}.")
-
-        {:reply, :ok, %{state | validators: Map.delete(validators, pubkey)}}
-
-      :error ->
+  def handle_call({:delete_validator, pubkey}, _from, %{validator_set: validator_set} = state) do
+    validator_set.validators
+    |> Enum.find(fn {_index, validator} -> validator.keystore.pubkey == pubkey end)
+    |> case do
+      {index, _validator} ->
+        Logger.warning("[Libp2pPort] Deleting validator with index #{inspect(index)}.")
+        updated_validators = Map.delete(validator_set.validators, index)
+        {:reply, :ok, Map.put(state.validator_set, :validators, updated_validators)}
+
```

Comment:
Maybe we can have a `ValidatorSet.delete(pubkey)`?

Reply:
Sure, I'll added this to #1281 and will tackle it soon!